kernel/kernel-api/src/main/java/io/delta/kernel/TableIdentifier.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel;
+
+/** Identifier for a table. e.g. <catalog> / <schema> / <tableName> */
+public class TableIdentifier {
+  /** The name of the table. */
+  private final String name;
+
+  /** The namespace of the table. e.g. <catalog> / <schema> */
+  private final String[] namespace;
+
+  public TableIdentifier(String[] namespace, String name) {
+    this.namespace = namespace;
+    this.name = name;
```

Comment:
`requireNonNull` seems missing. Also, it would be nice to ensure non-empty in my opinion.